lib/ecmarkdown.js

```diff
@@ -1,11 +1,357 @@
 'use strict';
-var parser = require('./generated-parser.js');
+
 var escapeHtml = require('escape-html');
+var beautify = require('./beautify.js');
 
-exports.list = function (ecmarkdown) {
-    return parser.parse(ecmarkdown.trim() + '\n', { startRule: 'list' });
+exports.document = function (ecmarkdown) {
+  return beautify(emitDocument(parse(ecmarkdown)));
 };
 
 exports.fragment = function (ecmarkdown) {
-    return parser.parse(escapeHtml(ecmarkdown.trim()), { startRule: 'fragment' });
+  return beautify(emitFragment(parse(escapeHtml(ecmarkdown))));
 };
+
+
+var lexemes = {
+  star: /\*/,
+  underscore: /_/,
+  tick: /`[^`\n]*`/,
```

Comment:
Add a comment explaining why star, underscore, and tilde get a simpler treatment than tick, pipe, and string? Or we could get rid of the difference, if that doesn't break things horribly?